arch/arm/core/aarch32/swap_helper.S

```diff
@@ -201,6 +197,11 @@ out_fp_endif:
     str r0, [r4]
 #endif
 
+#endif
+
+#if defined(CONFIG_ARM_STORE_EXC_RETURN)
+    /* Restore EXC_RETURN value. */
+    ldrsb lr, [r2, #_thread_offset_to_mode_exc_return]
```

Comment:
@oyvindronningstad just a question - why this is LDRSB, and not LDRB?

Reply:
LDRSB does sign extension which gives the initial 0xFFFFFF.